### DISCUSSION TITLE
Problem DNS

### DISCUSSION THREAD

**Discussion by xgusto** [eduardogsilva/wireguard_webadmin#34] [Help and Support]
The discussion started here https://github.com/eduardogsilva/wireguard_webadmin/discussions/22#discussioncomment-9965923
docker dns seems to be running
```
root@e914f6768e8a:/app# ping wireguard-webadmin-dns
PING wireguard-webadmin-dns (172.18.0.3): 56 data bytes
64 bytes from 172.18.0.3: icmp_seq=0 ttl=64 time=0.072 ms
64 bytes from 172.18.0.3: icmp_seq=1 ttl=64 time=0.166 ms
64 bytes from 172.18.0.3: icmp_seq=2 ttl=64 time=0.128 ms
64 bytes from 172.18.0.3: icmp_seq=3 ttl=64 time=0.144 ms
64 bytes from 172.18.0.3: icmp_seq=4 ttl=64 time=0.142 ms
^C--- wireguard-webadmin-dns ping statistics ---
5 packets transmitted, 5 packets received, 0% packet loss
round-trip min/avg/max/stddev = 0.072/0.130/0.166/0.032 ms
```

**Comment by eduardogsilva** [comment]
That's interesting, can you please check the first lines of your wg firewall?

cat /etc/wireguard/wg-firewall.sh 
I'm looking for a small snippet on the first lines,

something like:
```

DNS_IP=$(getent hosts wireguard-webadmin-dns | awk '{ print $1 }')
if [ -z "$DNS_IP" ]; then
    DNS_IP="127.0.0.250"
fi


```

At this moment, I'm writing a diagnostic tool for testing the dns service and help on troubleshoot.

**Reply by eduardogsilva** [reply]
Can you please run this inside your container and also paste the output? 

`getent hosts wireguard-webadmin-dns | awk '{ print $1 }'`

**Comment by xgusto** [comment]
![iptables01](https://github.com/eduardogsilva/wireguard_webadmin/assets/53255570/f64dd83a-8561-4341-9e33-ab60f4c423d4)

**Reply by eduardogsilva** [reply]
Can you get the output from:

`getent hosts wireguard-webadmin-dns | awk '{ print $1 }'
`

Also, can you please go again to Status > Update wireguard configuration and then click on Restart wireguard service?

After doing this, please check again the iptables rules. (and print the output here please)

**Comment by eduardogsilva** [comment]
Hello @xgusto 
I made a small script for helping with this type of troubleshooting.

Can you please upgrade to version 0.9612? 
After the upgrade, go to Console > dns test

Please, paste the output here.

Cheers!

**Reply by xgusto** [reply]
> Hello @xgusto I made a small script for helping with this type of troubleshooting.
> 
> Can you please upgrade to version 0.9612? After the upgrade, go to Console > dns test
> 
> Please, paste the output here.
> 
> Cheers!

I will update later. I'll get back to you later, thanks.

**Comment by xgusto** [comment]
```
getent hosts wireguard-webadmin-dns | awk '{ print $1 }'
172.18.0.3
```

```
Chain WGWADM_PREROUTING (1 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 DNAT       17   --  wg0    *       0.0.0.0/0            10.10.3.1            udp dpt:53 to:172.18.0.3:53
    0     0 DNAT       6    --  wg0    *       0.0.0.0/0            10.10.3.1            tcp dpt:53 to:172.18.0.3:53
    0     0 DNAT       17   --  wg1    *       0.0.0.0/0            10.10.4.1            udp dpt:53 to:172.18.0.3:53
    0     0 DNAT       6    --  wg1    *       0.0.0.0/0            10.10.4.1            tcp dpt:53 to:172.18.0.3:53
```
BTW I don't need to deal with DNS, I'm fine with IP addresses. I just wanted to try DNS, if there is such a possibility.

**Reply by eduardogsilva** [reply]
Now, you have the correct DNS rules applied. It should be working fine without any other issues.
You can configure your peers to use 10.10.3.1 or 10.10.4.1 as DNS servers. 

If you still have issues, please bring me the output from the diagnostic tool.

Cheers!

**Comment by xgusto** [comment]
Here is the output from the dns test. The DNS name is still not resolved.
```
--- Ping the DNS container...
PING wireguard-webadmin-dns (172.18.0.4): 56 data bytes
64 bytes from 172.18.0.4: icmp_seq=0 ttl=64 time=0.324 ms
64 bytes from 172.18.0.4: icmp_seq=1 ttl=64 time=0.073 ms
64 bytes from 172.18.0.4: icmp_seq=2 ttl=64 time=0.070 ms
64 bytes from 172.18.0.4: icmp_seq=3 ttl=64 time=0.161 ms
--- wireguard-webadmin-dns ping statistics ---
4 packets transmitted, 4 packets received, 0% packet loss
round-trip min/avg/max/stddev = 0.070/0.157/0.324/0.103 ms


--- Checking firewall rules...
 pkts bytes target     prot opt in     out     source               destination         
    0     0 DNAT       17   --  wg0    *       0.0.0.0/0            10.10.3.1            udp dpt:53 to:172.18.0.4:53
    0     0 DNAT       6    --  wg0    *       0.0.0.0/0            10.10.3.1            tcp dpt:53 to:172.18.0.4:53
    0     0 DNAT       17   --  wg1    *       0.0.0.0/0            10.10.4.1            udp dpt:53 to:172.18.0.4:53
    0     0 DNAT       6    --  wg1    *       0.0.0.0/0            10.10.4.1            tcp dpt:53 to:172.18.0.4:53


--- Testing the DNS resolution...
Resolving google.com...
142.251.208.142


--- Testing getent hosts...
172.18.0.4      wireguard-webadmin-dns
DNS IP: 172.18.0.4
```

**Comment by eduardogsilva** [comment]
Hello @xgusto , sorry for taking long, but I was involved in another issue.

I made some tests here, and the dns resolution for clients are working fine with the current version 0.9612. 

Here is the test that I made: 


1- I've created a static dns entry: teste.local pointing to the IP 192.168.112.2
2- My client connected to the wireguard VPN using the private address 10.88.188.3

3- I've made a DNS query for the server's corresponding wg interface and got the expected result.

```
eduardo@hades:~$ dig teste.local @10.88.188.1 +short
192.168.112.2

```

If you don't have dig on your system, you could try using nslookup
```
eduardo@hades:~$ nslookup
> server 10.88.188.1
Default server: 10.88.188.1
Address: 10.88.188.1#53
> teste.local
Server:		10.88.188.1
Address:	10.88.188.1#53

Name:	teste.local
Address: 192.168.112.2
> 
```


Just remember, your the DNS server used for your client, must match the same wg network interface address.

cheers!
